test/e2e/pkg_test.go

```diff
@@ -224,3 +224,31 @@ func TestExternallyManagedServiceAccount(t *testing.T) {
 			Feature(),
 	)
 }
+
+func TestConfigurationWithDigest(t *testing.T) {
+	manifests := "test/e2e/manifests/pkg/configuration/digest"
+
+	environment.Test(t,
+		features.NewWithDescription(t.Name(), "Tests that a Configuration with digest which depends on a Provider with digest will become healthy when the Provider becomes healthy").
+			WithLabel(LabelArea, LabelAreaPkg).
+			WithLabel(LabelSize, LabelSizeSmall).
+			WithLabel(config.LabelTestSuite, config.TestSuiteDefault).
+			WithSetup("ApplyConfiguration", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "configuration.yaml"),
+				funcs.ResourcesCreatedWithin(1*time.Minute, manifests, "configuration.yaml"),
+			)).
+			Assess("ConfigurationIsHealthy",
+				funcs.ResourcesHaveConditionWithin(2*time.Minute, manifests, "configuration.yaml", pkgv1.Healthy(), pkgv1.Active())).
+			Assess("RequiredProviderIsHealthy",
+				funcs.ResourcesHaveConditionWithin(2*time.Minute, manifests, "provider-dependency.yaml", pkgv1.Healthy(), pkgv1.Active())).
```

Comment:
nit: Possibly we can revert the order of assesments here. Provider will become ready before configuration as configuration would need to wait for it.